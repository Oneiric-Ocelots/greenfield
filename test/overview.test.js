import React from "react";
import { mount, shallow, configure } from "enzyme";

import ProductDetails from "../src/components/ProductDetails/ProductDetails.jsx";

test("adds 1+2 to equal to 3", () => {
  expect(((a, b) => a + b)(1, 2)).toBe(3);
});
