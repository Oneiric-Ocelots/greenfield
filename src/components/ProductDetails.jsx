import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import Axios from "axios";

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    height: "800px"
  },
  paper: {
    height: "100px",
    padding: theme.spacing(`2`),
    textAlign: "center",
    color: theme.palette.text.secondary
  }
}));

const ProductDetailsNested = () => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Grid container spacing={2}>
        <Grid item xs={9}></Grid>
        <Grid item xs={3}></Grid>
      </Grid>
    </div>
  );
};

class ProductDetails extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    let productId = 1;
    const url = `http://18.223.1.30/products/${productId}/styles`;

    Axios.get(url).then(results => {
      console.log(results);
    });
  }

  render() {
    <ProductDetailsNested />;
  }
}

export default ProductDetails;
