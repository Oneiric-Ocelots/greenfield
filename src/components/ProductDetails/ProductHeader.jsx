import React from "react";
import { connect } from "react-redux";

const ProductHeader = ({ product, styles, styleIndex }) => {
  const styleName = styles[styleIndex] ? styles[styleIndex].name : "";
  return (
    <div>
      <p className="mockup">{product.category}</p>
      <h2>{product.name}</h2>
      <p className="mockup">{`$${product.default_price}`}</p>
      <p className="mockup">
        <b>STYLE ></b> {styleName}
      </p>
    </div>
  );
};

export default connect(state => ({
  product: state.productInfo,
  styles: state.productStyles,
  styleIndex: state.selectedStyleIndex
}))(ProductHeader);
