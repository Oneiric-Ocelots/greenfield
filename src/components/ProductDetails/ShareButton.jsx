import React from "react";
import trackElement from "../../../api/trackElement";
import Button from "@material-ui/core/Button";
import ShareIcon from "@material-ui/icons/Share";
import {
  TwitterShareButton,
  TwitterIcon,
  FacebookShareButton,
  FacebookIcon,
  PinterestShareButton,
  PinterestIcon
} from "react-share";

const ShareButton = ({ product, share, toggleShare, photo, classes }) => (
  <>
    <Button
      variant="outlined"
      className={classes.shareButton}
      label={"none"}
      onClick={e => {
        trackElement(e);
        toggleShare(!share);
      }}
      aria-label="share"
    >
      <ShareIcon fontSize={"small"} />
    </Button>
    <div className={share ? "share-container" : "hidden2"}>
      <TwitterShareButton
        url={window.location.href}
        title={`Check out this cool ${product.name} on ShopBox. #ShopBox #${product.category}`}
        className="share-button"
        onClick={trackElement}
      >
        <TwitterIcon size={32} round />
      </TwitterShareButton>
      <FacebookShareButton
        url={window.location.href}
        quote={`Check out this cool ${product.name} on ShopBox. #ShopBox #${product.category}`}
        className="share-button"
        onClick={trackElement}
      >
        <FacebookIcon size={32} round />
      </FacebookShareButton>
      <PinterestShareButton
        url={window.location.href}
        media={photo ? photo : ""}
        windowWidth={1000}
        windowHeight={730}
        className="share-button"
        onClick={trackElement}
      >
        <PinterestIcon size={32} round />
      </PinterestShareButton>
    </div>
  </>
);

export default ShareButton;
