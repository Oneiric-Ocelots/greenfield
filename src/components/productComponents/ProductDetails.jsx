import React from "react";

export default function ProductDetails() {
  return <div className="product-details">ProductDetails</div>;
}
