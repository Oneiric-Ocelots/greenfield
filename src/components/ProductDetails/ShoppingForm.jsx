import React, { useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Button from "@material-ui/core/Button";
import EnabledQuantitySelector from "./EnabledQuantitySelector.jsx";
import DisabledQuantitySelector from "./DisabledQuantitySelector.jsx";
import InStockSizeForm from "./InStockSizeForm.jsx";
import ShareButton from "./ShareButton.jsx";
import trackElement from "../../../api/trackElement";

const useStyles = makeStyles(theme => ({
  root: {
    display: "flex",
    minWidth: "30px"
  },
  button: {
    margin: theme.spacing(1),
    marginLeft: "16px",
    width: "207px",
    height: "50px",
    border: "1px solid black",
    padding: "13.7px 10px 13.7px 20px",
    borderRadius: 0
  },
  shareButton: {
    margin: theme.spacing(1),
    width: "50px",
    height: "50px",
    border: "1px solid black",
    marginLeft: "16px",
    borderRadius: 0,
    color: "rgb(119,119,119)"
  },
  formControl: {
    margin: theme.spacing(1),
    minWidth: 120
  },
  selectSize: {
    margin: theme.spacing(1),
    width: "170px",
    height: "50px",
    border: "1px solid black",
    padding: "10px 10px 10px 20px"
  },
  labelInput: {
    padding: "12px 0 12px 12px",
    color: "rgba(0, 0, 0, 0.55) !important"
  },
  selectQuantity: {
    margin: theme.spacing(1),
    marginLeft: 0,
    width: "100px",
    height: "50px",
    border: "1px solid black",
    padding: "10px;"
  }
}));

export default function ShoppingForm({ style, product }) {
  // console.log("inside form", style);
  const classes = useStyles();
  const [quantity, setQuantity] = React.useState(1);
  const [size, setSize] = React.useState("");
  const [share, toggleShare] = React.useState(false);

  const handleChange = event => {
    if (event.target.name === "size") {
      setSize(event.target.value);
    } else if (event.target.name === "quantity") {
      setQuantity(event.target.value);
    }
  };

  const style_id = style ? style.style_id : 0;
  useEffect(() => {
    setQuantity(1);
    setSize("");
  }, [style_id]);

  const OutOfStockSizeForm = () => <></>;

  const inStock = style && Object.keys(style.skus);
  // FIXME: in stock is more complicated
  let maxQuantity = null;
  let quantityOptions = [];
  if (inStock) {
    maxQuantity = style.skus[size];
    maxQuantity = maxQuantity > 15 ? 15 : maxQuantity;
    if (maxQuantity) {
      for (let i = 0; i < maxQuantity; i++) {
        quantityOptions.push(i + 1);
      }
    }
  }

  if (inStock) {
    return (
      <div className="form-rows">
        <InStockSizeForm
          size={size}
          handleChange={handleChange}
          style={style}
          classes={classes}
        />
        {maxQuantity ? (
          <>
            <EnabledQuantitySelector
              quantities={quantityOptions}
              quantity={quantity}
              handleChange={handleChange}
              classes={classes}
            />
            <Button
              variant="outlined"
              className={classes.button}
              aria-label="Add to cart"
              onClick={trackElement}
            >
              Add to Cart +
            </Button>
            <ShareButton
              share={share}
              toggleShare={toggleShare}
              product={product}
              classes={classes}
              photo={style && style.photos[0].url}
            />
          </>
        ) : (
          <>
            <DisabledQuantitySelector classes={classes} />
            <Button
              variant="outlined"
              className={classes.button}
              disabled={true}
              aria-label="Add to cart"
            >
              Add to Cart +
            </Button>
            <ShareButton
              share={share}
              toggleShare={toggleShare}
              product={product}
              classes={classes}
              photo={style && style.photos[0].url}
            />
          </>
        )}
      </div>
    );
  } else {
    return (
      <div className="form-rows out-of-stock">
        <OutOfStockSizeForm />
        <Button variant="outlined" className={classes.button + " hidden"}>
          Add to Cart +
        </Button>
        <ShareButton
          share={share}
          toggleShare={toggleShare}
          product={product}
          classes={classes}
        />
      </div>
    );
  }
}
