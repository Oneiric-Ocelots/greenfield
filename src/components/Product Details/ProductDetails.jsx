import React, { useState, useEffect } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Grid from "@material-ui/core/Grid";
import ShoppingForm from "./ShoppingForm.jsx";
import ProductHeader from "./ProductHeader.jsx";
import StyleBubbleAreaContainer from "../../containers/StyleBubblesContainer";
import "../../styles.css";

const useStyles = makeStyles(theme => ({
  root: {
    flexGrow: 1,
    minWidth: "500px",
    maxWidth: "1100px"
  },
  defaultLeft: {
    height: "600px",
    overflow: "hidden"
  },
  defaultRight: {
    height: "600px",
    display: "flex",
    flexDirection: "column"
  }
}));

const ProductDetailsNested = ({ style }) => {
  const classes = useStyles();
  const imgUrl = style ? style.photos[0].url : "";
  return (
    <div className={classes.root}>
      <Grid container spacing={2}>
        <Grid item xs={8} className={classes.defaultLeft}>
          <img className="main-img" src={imgUrl}></img>
        </Grid>
        <Grid item xs={4} className={classes.defaultRight}>
          <div className="stars-mockup">stars and reviews link</div>
          <ProductHeader />
          <StyleBubbleAreaContainer />
          <ShoppingForm style={style} />
        </Grid>
      </Grid>
    </div>
  );
};

class ProductDetails extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    let id = window.location.pathname.split("/")[2];
    this.props.handleLoadProduct(id);
    this.props.handleLoadStyles(id);
    this.props.handleLoadMetadata(id);
  }

  render() {
    let style = this.props.styles
      ? this.props.styles[this.props.selectedStyleIndex]
      : null;
    return (
      <div className="product-details">
        <ProductDetailsNested style={style} />
      </div>
    );
  }
}

export default ProductDetails;
