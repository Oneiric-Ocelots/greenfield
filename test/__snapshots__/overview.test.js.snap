// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test that style bubble is connected to store Displays 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Memo(Connect(StyleBubble))
    style={
      Object {
        "photos": Array [
          Object {
            "thumbnail_url": "",
          },
        ],
      }
    }
  />
</Provider>
`;
